contracts/test/gateway/TestGatewayBase.sol

```diff
@@ -0,0 +1,179 @@
+    )
+        public
+        GatewayBase(
+            _stateRootProvider,
+            _bounty,
+            _organization
+        )
+    {}
+
+    /* external functions */
+
+    /**
+     * @notice It is used to set a message.
+     *
+     * @dev This is used for testing purpose.
+     *
+     * @param _intentHash Intent hash.
+     * @param _nonce Nonce of the message sender address.
+     * @param _gasPrice Gas price that message sender is ready to pay to
+     *                  transfer message.
+     * @param _gasLimit Gas limit that message sender is ready to pay.
+     * @param _sender Message sender address.
+     * @param _hashLock Hash Lock provided by the facilitator.
+     *
+     * @return messageHash_ Hash unique for every request.
+     */
+    function setMessage(
+        bytes32 _intentHash,
+        uint256 _nonce,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        address _sender,
+        bytes32 _hashLock
+    )
+        external
+        returns (bytes32 messageHash_)
+    {
+        MessageBus.Message memory message = getMessage(
+            _intentHash,
+            _nonce,
+            _gasPrice,
+            _gasLimit,
+            _sender,
+            _hashLock
+        );
+
+        messageHash_ = MessageBus.messageDigest(
+            message.intentHash,
+            message.nonce,
+            message.gasPrice,
+            message.gasLimit,
+            message.sender,
+            message.hashLock
+        );
+
+        messages[messageHash_] = message;
+
+    }
+
+    /**
+     * @notice It sets the status of inbox.
+     *
+     * @dev This is used for testing purpose.
+     *
+     * @param _messageHash It sets the status of the message.
+     * @param _status It sets the state of the message.
+     */
+    function setInboxStatus(
+        bytes32 _messageHash,
+        MessageBus.MessageStatus _status
+    )
+        external
+    {
+        messageBox.inbox[_messageHash] = _status;
+    }
+
+    /**
+     * @notice It sets the status of outbox.
+     *
+     * @dev This is used for testing purpose.
+     *
+     * @param _messageHash MessageHash for which status is the be set.
+     * @param _status Status of the message to be set.
+     */
+    function setOutboxStatus(
+        bytes32 _messageHash,
+        MessageBus.MessageStatus _status
+    )
+        external
+    {
+        messageBox.outbox[_messageHash] = _status;
+    }
+
+    /**
+     * @notice It sets the message hash for active inbox process.
+     *
+     * @dev This is used for testing purpose.
+     *
+     * @param _account Account address.
+     * @param _nonce Nonce of account address.
+     * @param _messageHash MessageHash for which status is the be set.
+     */
+    function setInboxProcess(
+        address _account,
+        uint256 _nonce,
+        bytes32 _messageHash
+    )
+        external
+    {
+        super.registerInboxProcess(_account, _nonce, _messageHash);
```

Comment:
Why it's important to have nonce in the argument? 
can we not directly set this?  
```
inboxActiveProcess[_account] = _messageHash;
```

Similar comment for setOutboxProcess